lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -621,3 +618,52 @@ def delete_stored_tls_resources(self):
             except OSError:
                 # thrown if file not exists, ignore
                 pass
+
+    def reload_tls_certificates(self):
+        """Reload transport and HTTP layer communication certificates via REST APIs."""
+        url_http = f"https://{self.charm.unit_ip}:9200/_plugins/_security/api/ssl/http/reloadcerts"
+        url_transport = (
+            f"https://{self.charm.unit_ip}:9200/_plugins/_security/api/ssl/transport/reloadcerts"
+        )
+
+        # using the SSL API requires authentication with app-admin cert and key
+        admin_secret = self.charm.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val) or {}
+
+        if not admin_secret:
+            logger.error("Can not update TLS certs, no app-admin secret found.")
+            return
+
+        admin_cert = f"{self.certs_path}/admin.cert"
+        admin_key = f"{self.certs_path}/admin.key"
+
+        with open(admin_cert, "w") as cert:
+            cert.write(admin_secret["cert"])
+
+        with open(admin_key, "w") as key:
+            key.write(admin_secret["key"])
+
+        try:
+            response_http = requests.put(
+                url_http,
+                cert=(admin_cert, admin_key),
+                verify=f"{self.certs_path}/chain.pem",
+                timeout=5,
+            )
+            response_transport = requests.put(
+                url_transport,
+                cert=(admin_cert, admin_key),
+                verify=f"{self.certs_path}/chain.pem",
+                timeout=5,
+            )
+            if not (
+                response_http.status_code == requests.codes.ok
+                and response_transport.status_code == requests.codes.ok
+            ):
```

Comment:
@Mehdi-Bendriss We've been wondering if it could be a good idea to extend the already existing `request()` method of Opensearch to handle this case (i.e. query with cert instead of admin credentials). What do you think? The use case is unique and out of ordinary, however may result in a more consistent usage of API requrests...

Reply:
Some context: I decided against extending the existing `request()` method because here is the only case where we need to authenticate against OpenSearch API with cert and key (instead of username/password).

If this should be added to the existing method, I'm fine with it. Would have been my approach as soon as there was another use case for cert/key authentication.

Reply:
Yes. The "pro" side was
 - extension on the existing, Opensearch internal `request()` method is not much
 - thus we have a homogenioius usage of `request()` across the Opensearch code.
 
 I'm fine with either solution, it's just worth a thought why we choose the one we do :-)

Reply:
Mehdi and I agreed on your proposal to extend the existing `request()` method for using certs instead of user/password for authentication.